### PR TITLE
Replace comments with Docstrings

### PR DESCRIPTION
**Pull request by JamesKohlsRepo**
Replaced comments with Docstrings, replying previous changes due to issues with main brach

### CODE REVIEW

File: mediabridge/data_processing/wiki_to_netflix.py [hunk 1]
```diff
@@ -190,4 +264,5 @@ def process_data(test=False):
     # Test is true if no argument is passed or if the first argument is not '--prod'.
     test = len(sys.argv) < 2 or sys.argv[1] != "--prod"
     process_data(test=test)
+    test = len(sys.argv) < 2 or sys.argv[1] != "--prod"
```
Comment (audiodude) [inline mediabridge/data_processing/wiki_to_netflix.py:267]: nit: Looks like a duplicate line from a merge? Please remove.

File: mediabridge/data_processing/wiki_to_netflix.py [hunk 2]
```diff
@@ -148,8 +217,13 @@ def wiki_query(data_csv, user_agent):
     return wiki_movie_ids, wiki_genres, wiki_directors
 
 
-# Calling all functions
 def process_data(test=False):
+    """
+    Calls all functions ...
+
+    Parameters:
+    test (bool): ...
```
Comment (audiodude) [inline mediabridge/data_processing/wiki_to_netflix.py:225]: Test is "If true, only process a subset of the data, for testing"

File: mediabridge/data_processing/wiki_to_netflix.py [hunk 3]
```diff
@@ -53,8 +98,17 @@ def wiki_feature_info(data, key):
...
 def format_sparql_query(title, year):
+    """
+    Formats SPARQL query for Wiki data
+
+    Parameters:
+    title (str): name of content to query
+    year (int): release year of the movie
+
+    Returns:
+    SPARKLE Query (str): formatted string with movie title and year
```
Comment (audiodude) [inline mediabridge/data_processing/wiki_to_netflix.py:110]: nit: SPARQL

File: mediabridge/data_processing/wiki_to_netflix.py [hunk 4]
```diff
@@ -29,14 +35,53 @@ def read_netflix_txt(txt_file, test):
...
+                    },
+                ]
+            }
+        }
+    key (str): The key for the information to extract (e.g. 'item', 'genreLabel', 'directorLabel').
+
+    Returns:
+        None: If the key is not present or no results are available
+        List: If the key is 'genreLabel', returns a list of unique genre labels.
+        String: If they Key is present, return the movie ID of the firt binding, in other words the first row in query result
```
Comment (audiodude) [inline mediabridge/data_processing/wiki_to_netflix.py:82]: nit: s/firt/first/

File: mediabridge/data_processing/wiki_to_netflix.py [hunk 5]
```diff
@@ -16,10 +16,16 @@ class WikidataServiceTimeoutException(Exception):
...
+    """
+    Reads and processes a Netflix text file.
+
+    Parameters:
+    txt_file (str): Path to the Netflix text file
+    test (Bool): When true, runs the functon in test mode
+    """
     num_rows = None
-    if test:
+    if test == True:
```
Comment (audiodude) [inline mediabridge/data_processing/wiki_to_netflix.py:28]: nit: bad merge? Please restore to `if test:`

**Review by audiodude — CHANGES_REQUESTED** [review]
Still looks great in it's new reincarnation, but I think there are some issues with the merge. Please double check the comments I've left, thanks!

**Comment by audiodude** [comment]
The ruff error can be fixed by applying the suggestions in my comments. Thanks!

File: mediabridge/data_processing/wiki_to_netflix.py (hunk 1 above)
Comment (JamesKohlsRepo) [reply mediabridge/data_processing/wiki_to_netflix.py:267]: Will do thanks for catching that

File: mediabridge/data_processing/wiki_to_netflix.py [hunk 6]
```diff
@@ -79,8 +79,7 @@ def wiki_feature_info(data, key):
     Returns:
         None: If the key is not present or no results are available
         List: If the key is 'genreLabel', returns a list of unique genre labels.
-        String: If they Key is present, return the movie ID of the firt binding, in other words the first row in query result
-
+        String: If they Key is present, return the movie ID of the first binding, in other words the first row in query result
```
Comment (audiodude) [inline mediabridge/data_processing/wiki_to_netflix.py:82]: nit: s/if they Key/if the Key/

File: mediabridge/data_processing/wiki_to_netflix.py [hunk 7]
```diff
@@ -148,8 +216,10 @@ def wiki_query(data_csv, user_agent):
     return wiki_movie_ids, wiki_genres, wiki_directors
 
 
-# Calling all functions
 def process_data(test=False):
+    """
+    If true, only process a subset of the data, for testing
```
Comment (audiodude) [inline mediabridge/data_processing/wiki_to_netflix.py:221]: how about:

```
Reads movie titles and release years from the Netflix data set (which should be downloaded and placed in the repo), then tries to match them with data from Wikidata. For any matches, a CSV file is written.
```

**Review by jhanley634 — APPROVED** [review]
Much of this looks good. Sorry about the merge nonsense, you'll need to tidy that up. Going forward, you (roughly) likely want to have at most two feature branches outstanding -- an old one that is out for review and a new one to work on while awaiting final review. The repo has a bunch of branches currently, which makes it a little harder to schedule work so no one steps on another's toes. Also, putting an issue number in the branch name (like "16-add-logging") can help to chronologically order them, and identify branches which have gone stale and maybe should be culled.

----

The wiki_feature_info() docstring is harder to read than it ought to be. The example dict is great but it takes up a lot of vertical space. Better to break out `def get_example_response()` having a single statement, which will `return` a constant, a dict. Several good things would come of that:

1. Humans can easily see there's a pair of input params, in a tabular list, readily scanned.
2. IDEs better support auto-indent when the material is code rather than plain text.
3. Unit tests can operate on the example record, without need of a network connection.

----

At the call site we see three parallel datastructures:
```
        wiki_movie_ids.append(wiki_feature_info(data, "item"))
        wiki_genres.append(wiki_feature_info(data, "genreLabel"))
        wiki_directors.append(wiki_feature_info(data, "directorLabel"))
```
It isn't always apparent to a maintenance engineer *whether*, and *how many* such lists are connected to one another. It is out-of-scope for this one, but in a subsequent PR you might consider defining a three-field [@dataclass](https://docs.python.org/3/library/dataclasses.html), to consolidate all this into just a single list. If for some reason you're not keen on that, there is always [NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple), or even the lowly dict.

**Review by audiodude — APPROVED** [review]

**Comment by audiodude** [comment]
I agree with @jhanley634 that parallel lists is an anti-pattern, see https://codeblog.jonskeet.uk/2014/06/03/anti-pattern-parallel-collections/

**Pull request merged**